File: source/basic/files/spectrum_sampling/spectrum_sampling/example02_fft.m
% ***************************************
%           Default FFT example
% ***************************************
% Ericsson Nikola Tesla
% DSP Hackaton 2021
% 
% Basic level example
%

close all;
clear all;

%% We run DTFT first
example01_dtft;

%% Find FFT

%
% NOTE
% We will now call `fft` with no additional parameters.
%
X_fft = fft(x);

% Bin axis
k = 0:length(X_fft)-1;

%% Plot the spectrum

figure(spectrumFig);

% Amplitude spectrum
subplot(2,1,1);
stem(2*pi*k/length(k), abs(X_fft), 'r');

% Phase spectrum
subplot(2,1,2);
stem(2*pi*k/length(k), angle(X_fft), 'r');

File: source/basic/files/spectrum_sampling/spectrum_sampling/example02_fft.py
# ***************************************
#           DTFT example
# ***************************************
# Ericsson Nikola Tesla
# DSP Hackaton 2021
# 
# Basic level example
#

import math
import numpy as np

import matplotlib.pyplot as plt

from scipy.fft import fft

## We run DTFT first
exec(open("./example01_dtft.py").read())

## Find FFT

#
# NOTE
# We will now call `fft` with no additional parameters.
#
X_fft = fft(x)

# Bin axis
k = np.arange(0,len(X_fft))

## Plot the spectrum
plt.figure(spectrumFig)

# Amplitude spectrum
plt.subplot(2,1,1)
plt.stem(2*math.pi*k/len(k), np.abs(X_fft), linefmt='r', basefmt=' ')

# Phase spectrum
plt.subplot(2,1,2)
plt.stem(2*math.pi*k/len(k), np.angle(X_fft), linefmt='r', basefmt=' ')

plt.show(block=False)

input('Example 02 complete.')
